### ISSUE TITLE
Upgrade to mssql-jdbc 7.4.1.jre8

### ISSUE THREAD

**Issue by boukewoudstra** [spring-projects/spring-boot#17965]
Is there any reason why latest stable Spring boot still uses `com.microsoft.sqlserver:mssql-jdbc:jar:6.4.0.jre8`? This dates back to February 2018.

Since then some issues have been fixed, for example `resultSet.getObject("dateTimeField", LocalDateTime.class)`. This otherwise gives an error:

```
com.microsoft.sqlserver.jdbc.SQLServerException: The conversion to class java.time.LocalDateTime is unsupported.
```

Upgrading to latest & greatest fixes the issue for me.  All it takes is a version bump to  `com.microsoft.sqlserver:mssql-jdbc:jar:7.4.1.jre8`

Thanks in advance for considering the upgrade.

**Comment by wilkinsona** [comment]
> Is there any reason why latest stable Spring boot still uses com.microsoft.sqlserver:mssql-jdbc:jar:6.4.0.jre8 ?

Yes. By default, we only pick up new major versions of dependencies in new major versions of Spring Boot. Spring Boot 2.0 shipped with 6.2.x so there hasn't been a new major version of Spring Boot that would naturally include a 7.x upgrade to SQL server's JDBC driver.

We can consider upgrading to 7.x in Spring Boot 2.2, but we'll need to be careful about how it may break users who are currently using 6.x.

**Comment by wilkinsona** [comment]
Looking at the [version compatibility table](https://docs.microsoft.com/en-us/sql/connect/jdbc/microsoft-jdbc-driver-for-sql-server-support-matrix?view=sql-server-2017#sql-version-compatibility), 7.4 of the JDBC driver adds support for SQL Server 2009 and drops support for SQL Server 2008 R2 which reached the end of extended support in July 2019. The [release notes](https://docs.microsoft.com/en-us/sql/connect/jdbc/release-notes-for-the-jdbc-driver?view=sql-server-2017) do not mention any breaking changes. From what I can tell, a jump to 7.x is reasonable for our 2.2 release.
